sympy/tensor/array/array_comprehension.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import print_function, division
+import functools
+from sympy.core.sympify import sympify
+from sympy.core.expr import Expr
+from sympy.core import Basic
+
+class ArrayComprehension(Basic):
+    """
+    Generate a list comprehension
+    """
+    def __new__(cls, expr, *values, **assumptions):
+        if any(len(l) != 3 or None for l in values):
+            raise ValueError('ArrayComprehension requires values lower and upper bound'
+                              'for the expression')
```

Comment:
give a space between `'` and `for`  like,  ' for the expression')

Reply:
I am sorry that I didn't see your review, I will uodate it in next commit. Thanks a lot!!